File: src/Sample/Flutter.Net/ScrollbarDemoPage.cs
using Avalonia;
using Avalonia.Media;
using Flutter.Rendering;
using Flutter.Widgets;

// Dart parity source (reference): dart_sample/lib/scrollbar_demo_page.dart (exact sample parity)

namespace Flutter.Net;

public sealed class ScrollbarDemoPage : StatefulWidget
{
    public override State CreateState()
    {
        return new ScrollbarDemoPageState();
    }
}

internal sealed class ScrollbarDemoPageState : State
{
    private ScrollController _controller = null!;

    public override void InitState()
    {
        _controller = new ScrollController();
    }

    public override void Dispose()
    {
        _controller.Dispose();
    }

    public override Widget Build(BuildContext context)
    {
        return new Column(
            crossAxisAlignment: CrossAxisAlignment.Stretch,
            spacing: 10,
            children:
            [
                new Text("Scrollbar + ScrollController", fontSize: 20, color: Colors.Black),
                new Text(
                    "The thumb is computed from viewport/content metrics.",
                    fontSize: 14,
                    color: Colors.DimGray),
                new Expanded(
                    child: new Scrollbar(
                        controller: _controller,
                        child: ListView.Builder(
                            itemCount: 70,
                            controller: _controller,
                            itemExtent: 40,
                            padding: new Thickness(10),
                            itemBuilder: (_, index) => new Container(
                                color: index % 2 == 0 ? Colors.White : Color.Parse("#FFF4F7FA"),
                                padding: new Thickness(10, 8),
                                child: new Text($"scroll row #{index}", fontSize: 13, color: Colors.Black)),
                            addAutomaticKeepAlives: false))),
            ]);
    }
}

File: dart_sample/lib/scrollbar_demo_page.dart
import 'package:flutter/material.dart';

class ScrollbarDemoPage extends StatefulWidget {
  const ScrollbarDemoPage({super.key});

  @override
  State<ScrollbarDemoPage> createState() => _ScrollbarDemoPageState();
}

class _ScrollbarDemoPageState extends State<ScrollbarDemoPage> {
  late final ScrollController _controller;

  @override
  void initState() {
    super.initState();
    _controller = ScrollController();
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.stretch,
      spacing: 10,
      children: <Widget>[
        const Text(
          'Scrollbar + ScrollController',
          style: TextStyle(fontSize: 20, color: Colors.black),
        ),
        const Text(
          'The thumb is computed from viewport/content metrics.',
          style: TextStyle(fontSize: 14, color: Colors.black54),
        ),
        Expanded(
          child: Scrollbar(
            controller: _controller,
            child: ListView.builder(
              controller: _controller,
              itemCount: 70,
              itemExtent: 40,
              padding: const EdgeInsets.all(10),
              itemBuilder: (BuildContext context, int index) {
                return Container(
                  color: index.isEven ? Colors.white : const Color(0xFFF4F7FA),
                  padding: const EdgeInsets.symmetric(
                    horizontal: 10,
                    vertical: 8,
                  ),
                  child: Text(
                    'scroll row #$index',
                    style: const TextStyle(fontSize: 13, color: Colors.black),
                  ),
                );
              },
            ),
          ),
        ),
      ],
    );
  }
}
